lib/std/child_process.zig

```diff
@@ -338,7 +338,10 @@ pub const ChildProcess = struct {
         }
 
         try child.spawn();
-        try child.collectOutput(&stdout, &stderr, args.max_output_bytes);
```

Comment:
Generally it is safer to do instead
```
errdefer child.kill() catch {};
try child.collectOutput(&stdout, &stderr, args.max_output_bytes);
```

Feel free to also cleanup `fn waitUnwrappedWindows(self: *ChildProcess) !void {`, which does not call
```zig
        os.close(self.id);
        os.close(self.thread_handle);
```
in all cases via `defer` ie if `GetProcessorInfo` fails.

Reply:
I did `errdefer child.kill() catch {};` at first there but that wont work since `stdout.toOwnedSlice()` may fail, then it will try to kill a dead process and wait on it which isn't safe

Reply:
I did the changes in waitUnwrappedWindows now.

Reply:
Not very important, but a block could be used to limit the scope of the `errdefer` if you wanted to still use `errdefer` (should be effectively equivalent to your current fix).

```zig
{
    try child.spawn();
    errdefer _ = child.kill() catch {};
    try child.collectOutput(&stdout, &stderr, args.max_output_bytes);
}
```

Reply:
Ah thats true, I did that now

Reply:
> im not 100% if zig guarantees struct initialization order

That is a good question, because it messes up the order of `try`, `defer` and `errdefer` and makes it arbitrary.